File: NeetCode150/Sliding Window/121. Best Time to Buy and Sell Stock/sol.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.empty()) return 0;
        int buy = prices[0];
        int profit = 0;
        int curPro = 0;

        for (int i=1; i<prices.size(); i++) {
            if (prices[i] < buy)    buy = prices[i];
            else {
                curPro = prices[i] - buy;
                profit = max(curPro, profit);
            }
        }
        return profit;
    }
};

File: NeetCode150/Sliding Window/121. Best Time to Buy and Sell Stock/sol.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = 101
        curProfit = 0
        profit = 0

        for i in prices:
            if i < buy:
                buy = i
            curProfit = i - buy
            profit = max(curProfit, profit)

        return profit